credentials/generate-revocation-set.py

```diff
@@ -347,6 +308,53 @@ def get_revocations_points_by_skid(self, issuer_subject_key_id) -> list[dict]:
 
         return response["pkiRevocationDistributionPointsByIssuerSubjectKeyID"]["points"]
 
+    def get_paa_cert(self, initial_cert: x509.Certificate) -> Optional[x509.Certificate]:
+        """Get the PAA certificate for the CRL Signer Certificate."""
+        issuer_name_b64 = get_issuer_b64(initial_cert)
+        akid = get_akid(initial_cert)
+        if akid is None:
+            return
+        paa_certificate = None
+        while not paa_certificate:
+            try:
+                issuer_certificate = None
+                response = None
+                akid_hex = akid.hex().upper()
+                if self.use_rest:
+                    akid_hex = ':'.join([akid_hex[i:i+2] for i in range(0, len(akid_hex), 2)])
+                    logging.debug(
+                        f"Fetching issuer from:{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}")
+                    response = requests.get(
+                        f"{self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}").json()
+                    if response["approvedCertificates"]["certs"][0]["pemCert"]:
+                        issuer_certificate = x509.load_pem_x509_certificate(
+                            bytes(response["approvedCertificates"]["certs"][0]["pemCert"], 'utf-8'))
+                    else:
+                        raise requests.exception.NotFound(
+                            f"No certificate found for {self.rest_node_url}/dcl/pki/certificates/{issuer_name_b64}/{akid_hex}")
+                else:
+                    query_cmd_list = ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex]
+
+                    response = self.get_dcld_cmd_output_json(
+                        ['query', 'pki', 'x509-cert', '-u', issuer_name_b64, '-k', akid_hex])
+                    if response["approvedCertificates"]["certs"][0]["pemCert"]:
+                        issuer_certificate = x509.load_pem_x509_certificate(
+                            bytes(response["approvedCertificates"]["certs"][0]["pemCert"], 'utf-8'))
+                    else:
+                        raise requests.exception.NotFound(f"No certificate found for dcl query{' '.join(query_cmd_list)}")
```

Comment:
Can you please abstract all the data-getting into a new interface (base class) + 2 concrete classes for dcld and HTTP REST respectively. This would make it easier to maintain and split the code of the different methods of getting the raw data (i.e. what is the input of `load_pem_x509_certificate`) into common code.

Reply:
Good thinking, that will be useful for the next change that allows for local files to be used. The interface has been added, the two classes, and the main impl updated to pick between the two. Thanks!